frontend/src/core/ai/context/providers/tables.ts

```diff
@@ -0,0 +1,212 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import type { Completion } from "@codemirror/autocomplete";
+import type { DatasetTablesMap } from "@/core/datasets/data-source-connections";
+import type { DataTable } from "@/core/kernel/messages";
+import type { AIContextItem } from "../registry";
+import { AIContextProvider } from "../registry";
+import { Boosts } from "./common";
+
+export interface TableContextItem extends AIContextItem {
+  type: "data";
+  data: DataTable;
+}
+
+export class TableContextProvider extends AIContextProvider<TableContextItem> {
+  readonly title = "Tables";
+  readonly mentionPrefix = "@";
+  readonly contextType = "data";
+
+  constructor(private tablesMap: DatasetTablesMap) {
+    super();
+  }
+
+  getItems(): TableContextItem[] {
+    return [...this.tablesMap.entries()].map(([tableName, table]) => ({
+      uri: this.asURI(tableName),
+      name: tableName,
+      type: this.contextType,
+      description: table.source === "memory" ? "in-memory" : table.source,
+      data: table,
+    }));
+  }
+
+  formatContext(item: TableContextItem): string {
+    const { data, uri: id } = item;
+    const { columns, source, num_rows, num_columns } = data;
+    const shape = [
+      num_rows == null ? undefined : `${num_rows} rows`,
+      num_columns == null ? undefined : `${num_columns} columns`,
+    ]
+      .filter(Boolean)
+      .join(", ");
+
+    let context = `Table: ${id}\nSource: ${source || "unknown"}`;
+    if (shape) {
+      context += `\nShape: ${shape}`;
+    }
+
+    if (columns && columns.length > 0) {
+      context += `\nColumns:\n${columns.map((col) => `  - ${col.name}: ${col.type}`).join("\n")}`;
+    }
+
+    return context;
+  }
+
+  getCompletions(): Completion[] {
+    return [];
+  }
+
+  formatCompletion(item: TableContextItem): Completion {
+    const tableName = item.data.name;
+    const table = item.data;
+    return {
+      label: `@${tableName}`,
+      displayLabel: tableName,
+      detail: table.source === "memory" ? "in-memory" : table.source,
+      boost:
+        table.source_type === "local"
+          ? Boosts.LOCAL_TABLE
+          : Boosts.REMOTE_TABLE,
+      type: table.variable_name ? "dataframe" : "table",
+      apply: `@${tableName}`,
+      section: table.variable_name ? "Dataframe" : "Table",
+      info: () => this.createTableInfoElement(tableName, table),
+    };
+  }
+
+  private createTableInfoElement(
+    tableName: string,
+    table: DataTable,
+  ): HTMLElement {
+    const infoContainer = document.createElement("div");
+    infoContainer.classList.add(
+      "mo-cm-tooltip",
+      "docs-documentation",
+      "min-w-[200px]",
+    );
+    infoContainer.style.display = "flex";
+    infoContainer.style.flexDirection = "column";
+    infoContainer.style.gap = ".8rem";
+    infoContainer.style.padding = "0.5rem";
```

Comment:
Any reason not to stick with tailwind?

```suggestion
    infoContainer.classList.add(
      "mo-cm-tooltip",
      "docs-documentation",
      "min-w-[200px]",
      "flex",
      "flex-col",
      "gap-[0.8rem]",
      "p-2"
    );
```